### PR TITLE
Create vm

### PR DESCRIPTION
**Pull request by LiYa-Xu**

### CODE REVIEW

**Comment by LiYa-Xu** [comment]
Add a new job in building stemcell pipeline to OS reload an existing VM with the newly created SL community stemcell.

File: ci/softlayer/tasks/deploy-vm.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export VERSION=$( cat version/number | sed 's/\.0$//;s/\.0$//' )
+
+echo "[INFO] Install dependencies packages"
+apt-get update
+apt-get install curl -y
+
+tar zxvf stemcell/*.tgz  -C ./
+image_id=`grep "virtual-disk-image-uuid:" stemcell.MF| cut -d ":" -f2 | sed 's/^[ \t]*//g' `
+
+curl -X POST -d '{"parameters":[ "FORCE",  { "imageTemplateId": "$image_id" }] }'https://$SL_USERNAME:$SL_API_KEY@api.softlayer.com/rest/v3/SoftLayer_Virtual_Guest/74649319/reloadOperatingSystem.json
```
Comment (mattcui) [inline ci/softlayer/tasks/deploy-vm.sh]: Should not hardcode vm id, and don't expose vm id in public.

**Review by mattcui — CHANGES_REQUESTED** [review]
Make sure to pass shell check for all shell scripts.

File: ci/softlayer/tasks/deploy-vm.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -e -x -u
```
Comment (mattcui) [inline ci/softlayer/tasks/deploy-vm.sh]: `set -o errexit -o nounset -o pipefail`

File: ci/softlayer/tasks/deploy-vm.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  - name: stemcells-index
+run:
+  path: bosh-linux-stemcell-builder/ci/softlayer/tasks/deploy-vm.sh
+
+params:
+  PUBLISHED_BUCKET_NAME: required
+  OS_NAME:               required
+  OS_VERSION:            required
+  SL_USER:               required
+  SL_TOKEN:              required
```
Comment (mattcui) [inline ci/softlayer/tasks/deploy-vm.yml]: should be `SL_API_KEY`??

File: ci/softlayer/tasks/deploy-vm.sh (hunk 1 above)
Comment (mattcui) [inline ci/softlayer/tasks/deploy-vm.sh]: need result check

**Review by mattcui — COMMENTED** [review]
@LiYa-Xu  Please respond and mark resolve for each my comment, thanks

**Comment by edwardstudy** [comment]
@mattcui @LiYa-Xu Could we merge this PR?

File: ci/softlayer/tasks/deploy-vm.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -o errexit -o nounset -o pipefail
+
```
Comment (edwardstudy) [inline ci/softlayer/tasks/deploy-vm.sh:4]: Could we follow other scripts and add `sed -e` to each script?

**Pull request merged**